### ISSUE TITLE
Broken integration tests for uniCOIL

### ISSUE THREAD

**Issue by lintool** [castorini/pyserini#1051]
@MXueguang can you please take a look?

```
======================================================================
ERROR: test_msmarco_doc_unicoil_d2q_otf (test_unicoil.TestSearchIntegration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/jimmylin/pyserini/integrations/sparse/test_unicoil.py", line 69, in test_msmarco_doc_unicoil_d2q_otf
    self.assertAlmostEqual(score, 0.3531, delta=0.0001)
  File "/opt/local/stow/Python3-3.8.12/lib/python3.8/unittest/case.py", line 943, in assertAlmostEqual
    diff = abs(first - second)
TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'

======================================================================
ERROR: test_msmarco_passage_unicoil_d2q_otf (test_unicoil.TestSearchIntegration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/jimmylin/pyserini/integrations/sparse/test_unicoil.py", line 51, in test_msmarco_passage_unicoil_d2q_otf
    self.assertAlmostEqual(score, 0.3509, delta=0.0001)
  File "/opt/local/stow/Python3-3.8.12/lib/python3.8/unittest/case.py", line 943, in assertAlmostEqual
    diff = abs(first - second)
TypeError: unsupported operand type(s) for -: 'NoneType' and 'float'

----------------------------------------------------------------------
```

**Comment by lintool** [comment]
Fixed by #1052

**Comment by lintool** [comment]
@MXueguang additional broken tests:

```
======================================================================
ERROR: test_unicoil_encode_as_jsonl (test_encode.TestSearchIntegration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/jimmylin/pyserini/integrations/dense/test_encode.py", line 80, in test_unicoil_encode_as_jsonl
    searcher = LuceneImpactSearcher(index_dir, query_encoder='castorini/unicoil-d2q-msmarco-passage')
  File "/scratch2/jimmylin/pyserini/pyserini/search/lucene/_impact_searcher.py", line 51, in __init__
    self.idf = self._compute_idf(index_dir)
  File "/scratch2/jimmylin/pyserini/pyserini/search/lucene/_impact_searcher.py", line 236, in _compute_idf
    index_reader = IndexReader(index_path)
  File "/scratch2/jimmylin/pyserini/pyserini/index/_base.py", line 187, in __init__
    self.reader = self.object.getReader(index_dir)
  File "jnius/jnius_export_class.pxi", line 884, in jnius.JavaMethod.__call__
  File "jnius/jnius_export_class.pxi", line 1056, in jnius.JavaMethod.call_staticmethod
  File "jnius/jnius_utils.pxi", line 91, in jnius.check_exception
jnius.JavaException: JVM exception occurred: no segments* file found in MMapDirectory@/scratch2/jimmylin/pyserini/temp_lucene lockFactory=org.apache.lucene.store.NativeFSLockFactory@528c868: files: [] org.apache.lucene.index.IndexNotFoundException

======================================================================
ERROR: test_unicoil_encode_as_jsonl (test_encode.TestSearchIntegration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/jimmylin/pyserini/integrations/dense/test_encode.py", line 88, in tearDown
    shutil.rmtree(f)
  File "/scratch2/jimmylin/virtualenv/pyserini-dev/lib/python3.8/shutil.py", line 709, in rmtree
    onerror(os.lstat, path, sys.exc_info())
  File "/scratch2/jimmylin/virtualenv/pyserini-dev/lib/python3.8/shutil.py", line 707, in rmtree
    orig_st = os.lstat(path)
FileNotFoundError: [Errno 2] No such file or directory: './temp_embeddings'

----------------------------------------------------------------------
```
